### ISSUE TITLE
Possible TypeScript type definitions regression on 1.2.3

### ISSUE THREAD

**Issue by AlexTMjugador** [shishkin/astro-pagefind#30]
Hi, thanks for your work on this useful Pagefind integration for Astro!

I use it in an open-source project where Renovate automatically updates dependencies, and as a result I have noticed that bumping `astro-pagefind` from 1.2.3 to 1.2.3 caused some new TypeScript checking errors, which you can read in [this CI workflow run log](https://github.com/AlexTMjugador/Home/actions/runs/5631026797/job/15257586471?pr=33).

Since this is a patch release, there should not be any breaking changes like this. As far as I know, my project isn't doing anything wrong with how it integrates `astro-pagefind`, but I'd love to hear if I'm wrong or should change something on my end.

**Comment by akyag** [comment]
@shishkin Hello, thanks for working on bringing Pagefind to Astro, This is my first time using it and I am pretty excited to finally have a search for a static website. Once your PR gets merged, this will be fixed right? I can help with testing. Also if I used the previous version, is that currently working?

**Comment by shishkin** [comment]
@AlexTMjugador do you still see the issue after upgrading Astro to [2.10.8](https://github.com/withastro/astro/releases/tag/astro%402.10.8)?

**Comment by shishkin** [comment]
> How do I pass arguments like --bundle-dir mysearch for example. Need that for using GitHub pages as a host.

@akyag You should not need to change pagefind's bundle dir for that. Take a look at [this Astro guide](https://docs.astro.build/en/guides/deploy/github/).

**Comment by AlexTMjugador** [comment]
> @AlexTMjugador do you still see the issue after upgrading Astro to [2.10.8](https://github.com/withastro/astro/releases/tag/astro%402.10.8)?

Yes, the problem still occurs with Astro 2.10.8 and later. Specifically, I tested with 2.10.15. The errors haven't changed for me since I reported this issue.

**Comment by AlexTMjugador** [comment]
Updating Astro to 3.1.0 fixed this issue for me :tada:
